### PR TITLE
Impose 1000 character limit for search terms

### PR DESCRIPTION
**Pull request by bh412**
---

**Pull Request Check List**
- [ ] Unit Tests & Regression Tests Added (Optional)
- [x] Removed Unnecessary Logs/System.Outs/Comments/TODOs
- [ ] Added enough Logging to monitor expected behaviour change
- [ ] Security - Injection - everything run by an interpreter (SQL, OS...) is either validated or escaped
- [x] Security - Data Exposure - PII is not stored or sent unencrypted
- [ ] Security - Data Exposure - Test any altered or created endpoints using [swagger](http://localhost:8080/isaac-api/api-docs/)
- [ ] Security - Access Control - Check authorisation on every new endpoint
- [ ] Security - New dependency - configured sensibly not relying on defaults
- [ ] Security - New dependency - Searched for any know vulnerabilities
- [ ] Security - New dependency - Signed up team to mailing list
- [ ] Security - New dependency - Added to dependency list
- [ ] DB schema changes - postgres-rutherford-create-script updated
- [ ] DB schema changes - upgrade script created matching create script
- [ ] Updated Release Procedure & Documentation (& Considered Implications to Previous Versions)
- [ ] Peer-Reviewed

### CODE REVIEW

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java [hunk 1]
```diff
@@ -189,9 +189,12 @@ public final Response search(@Context final Request request, @Context final Http
             @DefaultValue(DEFAULT_START_INDEX_AS_STRING) @QueryParam("start_index") final Integer startIndex,
             @DefaultValue(DEFAULT_SEARCH_RESULT_LIMIT_AS_STRING) @QueryParam("limit") final Integer limit) {
 
+        // Impose 1000 character search string limit internally
+        String internalSearchString = searchString.substring(0, Math.min(searchString.length(), 1000));
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java:193]: I think this should be a constant like `SEARCH_TEXT_CHAR_LIMIT` of 1000, and then if the `searchString` is longer than that we return a 400 Bad Request and not truncate it.
The front end can do something nicer like warn you, and have an even shorter limit.

Comment (bh412) [reply src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java:193]: Ahh I thought the idea was to silently truncate - I'll alter this

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java [hunk 2]
```diff
@@ -189,6 +189,12 @@ public final Response search(@Context final Request request, @Context final Http
             @DefaultValue(DEFAULT_START_INDEX_AS_STRING) @QueryParam("start_index") final Integer startIndex,
             @DefaultValue(DEFAULT_SEARCH_RESULT_LIMIT_AS_STRING) @QueryParam("limit") final Integer limit) {
 
+        // Impose 1000 character search string limit internally
+        if (searchString.length() > SEARCH_TEXT_CHAR_LIMIT) {
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java:193]: Does IntelliJ think this is safe? Is it that it wouldn't have matched the endpoint if searchString was null, and so you don't need a null check?

Comment (bh412) [reply src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java:193]: It infers the @notnull annotation - searchString has to be a string or it throws a 'can't find resource' when you try and hit it

**Review by jsharkey13 — APPROVED** [review]
Is there a corresponding front end PR to ensure this can't be hit by accident and that the error messages shown are obviously visible and explain the issue?

**Comment by bh412** [comment]
I've added a front end PR with a search box max length of 255 characters. The returned toast seems clear enough to me

**Pull request merged**